server/src/main/java/org/elasticsearch/common/bytes/ByteBufferReference.java

```diff
@@ -46,6 +46,20 @@ public byte get(int index) {
         return buffer.get(index);
     }
 
+    @Override
+    public int get(int index, byte[] buf, int offset, int length) {
+        if (length == 0) {
+            return 0;
+        }
+        buffer.position(index);
```

Comment:
This is not thread safe. It feels like a method that `get` bytes from an immutable BytesReference should not break if multiple callers call it.

Reply:
Fair point, neither is the `slice` on this thing though that's why I went with it. But I think this is just dead code anyway so I opened https://github.com/elastic/elasticsearch/pull/61116 :)